basiclibrary/src/test/java/basiclibrary/ArrayManipulationTest.java

```diff
@@ -0,0 +1,84 @@
+package basiclibrary;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class ArrayManipulationTest {
+
+    //test cases for roll function
+    @Test
+    public void testRollFunctionLength() {
+        ArrayManipulation classUnderTest = new ArrayManipulation();
+        int n = 4;
+        int[] expectedOutput = new int[4];
+        assertEquals("this should return 'true'", expectedOutput.length, classUnderTest.roll(n).length);
```

Comment:
So this isn't actually returning true which you state in the assert description. ClassUnderTest.roll(n).length is actually returning an int, which you are making sure it is the int you are expecting